src/main/java/uk/gov/pay/directdebit/payments/params/DirectDebitEventSearchParams.java

```diff
@@ -7,45 +7,46 @@
 import java.time.ZonedDateTime;
 import java.time.format.DateTimeFormatter;
 import java.time.format.DateTimeParseException;
-import java.util.HashMap;
+import java.util.LinkedHashMap;
 import java.util.Map;
 
 @Builder
 public class DirectDebitEventSearchParams {
     @Getter private final ZonedDateTime beforeDate;
     @Getter private final ZonedDateTime afterDate;
-    @Getter private final Long mandateId;
-    @Getter private final Long transactionId;
+    @Getter private final String mandateExternalId;
+    @Getter private final String transactionExternalId;
     @Getter private final Integer pageSize;
     @Getter private final Integer page;
 
     public DirectDebitEventSearchParamsBuilder copy() {
-        return new DirectDebitEventSearchParamsBuilder().page(page).pageSize(pageSize).transactionId(transactionId).mandateId(mandateId).afterDate(afterDate).beforeDate(beforeDate);
+        return new DirectDebitEventSearchParamsBuilder().page(page).pageSize(pageSize).transactionExternalId(transactionExternalId).mandateExternalId(mandateExternalId).afterDate(afterDate).beforeDate(beforeDate);
     }
 
     public Map<String, String> getParamsAsMap() {
-        Map<String, String> params = new HashMap<>();
+        Map<String, String> params = new LinkedHashMap<>();
```

Comment:
Why do we want to preserve the order here?

Reply:
It's not really required from a production-code perspective, but it was to make the tests in DirectDebitEventsPaginationTest pass as we do something like
```
assertEquals(PaginationLink.ofValue(testUrl + "/test/?before=2018-06-29T09%3A00%3A00Z&after=2018-06-29T08%3A00%3A00Z&mandate_external_id=1234L&transaction_external_id=5678L&page=2&page_size=10")
                , pagination.getSelfLink());
```
Perhaps we should perform the assertion in a better way that makes the ordering irrelevant?

Reply:
I thought it would be something to do with tests. I can see why you did this. Improving the assertion is a pain (you’d probably have to reparse the URL into a set or something) but could probably be reused everywhere we build up URLs like this. Keeping the order deterministic is also a reasonable solution. I’ll leave this one up to you.

Reply:
let's keep it deterministic for now then